src/main/kotlin/com/coxautodev/graphql/tools/SchemaParserBuilder.kt

```diff
@@ -280,7 +284,31 @@ data class SchemaParserOptions internal constructor(val genericWrappers: List<Ge
         }
     }
 
-    data class GenericWrapper(val type: Class<*>, val index: Int) {
-        constructor(type: KClass<*>, index: Int): this(type.java, index)
+    data class GenericWrapper(val type: Class<*>, val index: Int, val transformer: (Any, DataFetchingEnvironment) -> Any?) {
+        
+        constructor(type: Class<*>, index: Int): this(type, index, { x, _ -> x })
+        constructor(type: KClass<*>, index: Int): this(type.java, index, { x, _ -> x })
+        
+        companion object {
+
+            @Suppress("UNCHECKED_CAST")
+            @JvmStatic fun <T> withTransformer(type: Class<T>, index: Int, transformer: (T, DataFetchingEnvironment) -> Any?): GenericWrapper where T: Any {
+                return GenericWrapper(type, index, transformer as (Any, DataFetchingEnvironment) -> Any?)
+            }
+            
+            fun <T> withTransformer(type: KClass<T>, index: Int, transformer: (T, DataFetchingEnvironment) -> Any?): GenericWrapper where T: Any {
+                return withTransformer(type.java, index, transformer)
+            }
+
+            @JvmStatic fun <T> withTransformer(type: Class<T>, index: Int, transformer: (T) -> Any?): GenericWrapper where T: Any {
+                return withTransformer(type, index, { x, _ -> transformer.invoke(x) })
+            }
+
+            fun <T> withTransformer(type: KClass<T>, index: Int, transformer: (T) -> Any?): GenericWrapper where T: Any {
+                return withTransformer(type.java, index, transformer)
+            }
+            
+        }
```

Comment:
Java-style overloading, not sure whether Kotlin proposes some less verbose alternative.